lib/command/base.rb

```diff
@@ -38,16 +38,27 @@ class Base # rubocop:disable Metrics/ClassLength
     WITH_INFO_HEADER = true
     # Which validations to run before the command
     VALIDATIONS = %w[config].freeze
+    SUBCOMMAND = nil
 
     def initialize(config)
       @config = config
     end
 
-    def self.all_commands
-      Dir["#{__dir__}/*.rb"].each_with_object({}) do |file, result|
-        filename = File.basename(file, ".rb")
-        classname = File.read(file).match(/^\s+class (\w+) < Base($| .*$)/)&.captures&.first
-        result[filename.to_sym] = Object.const_get("::Command::#{classname}") if classname
+    def self.all_commands # rubocop:disable Metrics/MethodLength
```

Comment:
No, we're not defining the commands in each command file, we're just configuring them. Thor doesn't handle it like that. So we need to load all of the configurations and manually define the Thor commands, which is done in: https://github.com/shakacode/control-plane-flow/blob/66e532bf69e2d2136b15b3608eb9c5007120a0fb/lib/cpflow.rb#L168

Reply:
It seems that we need this method to be able to use classes i/o methods. In order to register a new coomand we need to do something like this:
```
class SomeClass < Thor
  desc "usage", "desc"
  long_desc "long desc"
  def command_name
  end
end
```
But we are creating separate classes for each command: 
```
module Command
  class SomeCommand < Base
    NAME = "command_name"
    USAGE = "usage"
    DESCRIPTION = "desc"
    LONG_DESCRIPTION = "long desc"

    def call
      ...
    end
  end
end
```

Which later transform to methods:
```
class SomeClass < Thor
  all_commands.each do |command_key, command_class|
    name = command_class::NAME
    usage = command_class::USAGE
    description = command_class::DESCRIPTION
    long_description = command_class::LONG_DESCRIPTION

    desc(usage, description, hide: hide)
    long_desc(long_description)
    define_method(name) do
      command_class.new.call
    end
  end
end
```

Reply:
@rafaelgomesxyz Why do we need to do it that way instead of reopening the main class and load all files on the [entrypoint](https://github.com/shakacode/control-plane-flow/blob/main/lib/cpflow.rb)?

This way each command would be declared normally like Thor expect us to do.